docs/source/tertiary/whole-genome-regression.rst

```diff
@@ -380,6 +382,19 @@ Example
     import math
     assert math.isclose(y_hat_df.at[('HG00096', '22'),'Continuous_Trait_1'], -0.5577744539844645)
 
+---------------
+Troubleshooting
+---------------
+
+If you encounter limits related to memory allocation in PyArrow, you may need to tune the number of alphas, number of
+variants per block, and/or the number of sample blocks. The default values for these hyperparameters are tuned for
+500,000 variants and 500,000 samples.
+
+
+The known limit is:
+``(# alphas) * (# variants/ # variants per block) * (# samples / # sample blocks) <= 132,152,839``
```

Comment:
Isn't there another limit based on the total number of header blocks?

Reply:
The total number of header blocks meaning `(# variants/ # variants per block) * (# sample blocks)`? I haven't encountered bottlenecks during the ridge regression (versus the reduction) step during my testing. This is the limit for any Arrow vector, so we can check that any assembled group during GloWGR is smaller than this scale.

Reply:
I see, actually the expression here does depend on the total number of header blocks as you say. Is that definitely during the reduction step? I thought that during reduction we never load in more than one block of the matrix at a time, so the total number of variants should not matter.

Also, I'm very surprised that the number of labels is not in this expression.

Reply:
Sorry, I wrote my last statement incorrectly - this is a bottleneck I encountered during the ridge regression step. In particular, it's in the map step after grouping by sample block and label (hence why the number of labels doesn't matter): https://github.com/projectglow/glow/blob/50afeadb5d980c7216517e1ce7c8572b467d3d02/python/glow/wgr/linear_model/ridge_model.py#L237

In this case, the number of variant blocks (`# variants/ # variants per block`) does matter.